pkg/operator/rules.go

```diff
@@ -109,6 +109,7 @@ func (prs *PrometheusRuleSelector) generateRulesConfiguration(promRule *monitori
 func (prs *PrometheusRuleSelector) sanitizePrometheusRulesSpec(promRuleSpec monitoringv1.PrometheusRuleSpec, logger *slog.Logger) monitoringv1.PrometheusRuleSpec {
 	minVersionKeepFiringFor := semver.MustParse("2.42.0")
 	minVersionLimits := semver.MustParse("2.31.0")
+	minVersionQueryOffset := semver.MustParse("2.53.0")
```

Comment:
(nit) just to make it clear, it would be nice to add `minVersionQueryOffset = semver.MustParse("100.0.0") // Arbitrary very high major version because it's not yet supported by Thanos.` when the format is Thanos.